Splitting by student and profession

This is split 0

In [1]:
import pandas as pd

In [2]:
org_data = pd.read_csv("../original_data/train.csv")


In [3]:
org_data['Working Professional or Student'].value_counts()


Working Professional or Student
Working Professional    112799
Student                  27901
Name: count, dtype: int64

In [4]:
prof_data = org_data.loc[org_data['Working Professional or Student'] == 'Working Professional']
student_data = org_data.loc[org_data['Working Professional or Student'] == 'Student']

Need to drop columns that no longer apply

In [5]:
prof_data.head()

id      Name  Gender   Age       City Working Professional or Student  \
0   0  Aaradhya  Female  49.0   Ludhiana            Working Professional   
1   1     Vivan    Male  26.0   Varanasi            Working Professional   
3   3    Yuvraj    Male  22.0     Mumbai            Working Professional   
4   4      Rhea  Female  30.0     Kanpur            Working Professional   
5   5      Vani  Female  59.0  Ahmedabad            Working Professional   

            Profession  Academic Pressure  Work Pressure  CGPA  \
0                 Chef                NaN            5.0   NaN   
1              Teacher                NaN            4.0   NaN   
3              Teacher                NaN            5.0   NaN   
4     Business Analyst                NaN            1.0   NaN   
5  Finanancial Analyst                NaN            2.0   NaN   

   Study Satisfaction  Job Satisfaction     Sleep Duration Dietary Habits  \
0                 NaN               2.0  More than 8 hours        Healthy   
1                 NaN               3.0  Less than 5 hours      Unhealthy   
3                 NaN               1.0  Less than 5 hours       Moderate   
4                 NaN               1.0          5-6 hours      Unhealthy   
5                 NaN               5.0          5-6 hours        Healthy   

  Degree Have you ever had suicidal thoughts ?  Work/Study Hours  \
0    BHM                                    No               1.0   
1    LLB                                   Yes               7.0   
3    BBA                                   Yes              10.0   
4    BBA                                   Yes               9.0   
5    MCA                                    No               7.0   

   Financial Stress Family History of Mental Illness  Depression  
0               2.0                               No           0  
1               3.0                               No           1  
3               1.0                              Yes           1  
4               4.0                              Yes           0  
5               5.0                               No           0

Even though there are still some values left here, these are possibly errounous captures or outliers

In [6]:
prof_data[~prof_data['Study Satisfaction'].isna()].index

Index([18752, 41915, 55827, 60348, 99062, 101189], dtype='int64')

In [7]:
prof_data['Academic Pressure'].value_counts()

Academic Pressure
4.0    3
2.0    1
3.0    1
Name: count, dtype: int64

In [8]:
prof_data_irrelevant_columns = ['Working Professional or Student','Academic Pressure','CGPA','Study Satisfaction']
prof_data = prof_data.drop(columns=prof_data_irrelevant_columns,axis=1)

In [9]:
prof_data.head()

id      Name  Gender   Age       City           Profession  Work Pressure  \
0   0  Aaradhya  Female  49.0   Ludhiana                 Chef            5.0   
1   1     Vivan    Male  26.0   Varanasi              Teacher            4.0   
3   3    Yuvraj    Male  22.0     Mumbai              Teacher            5.0   
4   4      Rhea  Female  30.0     Kanpur     Business Analyst            1.0   
5   5      Vani  Female  59.0  Ahmedabad  Finanancial Analyst            2.0   

   Job Satisfaction     Sleep Duration Dietary Habits Degree  \
0               2.0  More than 8 hours        Healthy    BHM   
1               3.0  Less than 5 hours      Unhealthy    LLB   
3               1.0  Less than 5 hours       Moderate    BBA   
4               1.0          5-6 hours      Unhealthy    BBA   
5               5.0          5-6 hours        Healthy    MCA   

  Have you ever had suicidal thoughts ?  Work/Study Hours  Financial Stress  \
0                                    No               1.0               2.0   
1                                   Yes               7.0               3.0   
3                                   Yes              10.0               1.0   
4                                   Yes               9.0               4.0   
5                                    No               7.0               5.0   

  Family History of Mental Illness  Depression  
0                               No           0  
1                               No           1  
3                              Yes           1  
4                              Yes           0  
5                               No           0

Look into the possibility of analysing these guys as students: 


prof_data[~prof_data['Study Satisfaction'].isna()]


Index([18752, 41915, 55827, 60348, 99062, 101189], dtype='int64')

In [10]:
prof_data.to_csv('../data_split_0/prof_train_data.csv',index=False)

In [11]:
student_data.head()

id       Name  Gender   Age           City  \
2    2     Yuvraj    Male  33.0  Visakhapatnam   
8    8  Aishwarya  Female  24.0      Bangalore   
26  26     Aditya    Male  31.0       Srinagar   
30  30     Prisha  Female  28.0       Varanasi   
32  32     Chhavi  Female  25.0         Jaipur   

   Working Professional or Student Profession  Academic Pressure  \
2                          Student        NaN                5.0   
8                          Student        NaN                2.0   
26                         Student        NaN                3.0   
30                         Student        NaN                3.0   
32                         Student        NaN                4.0   

    Work Pressure  CGPA  Study Satisfaction  Job Satisfaction  \
2             NaN  8.97                 2.0               NaN   
8             NaN  5.90                 5.0               NaN   
26            NaN  7.03                 5.0               NaN   
30            NaN  5.59                 2.0               NaN   
32            NaN  8.13                 3.0               NaN   

       Sleep Duration Dietary Habits   Degree  \
2           5-6 hours        Healthy  B.Pharm   
8           5-6 hours       Moderate      BSc   
26  Less than 5 hours        Healthy       BA   
30          7-8 hours       Moderate      BCA   
32          5-6 hours       Moderate   M.Tech   

   Have you ever had suicidal thoughts ?  Work/Study Hours  Financial Stress  \
2                                    Yes               3.0               1.0   
8                                     No               3.0               2.0   
26                                    No               9.0               1.0   
30                                   Yes               4.0               5.0   
32                                   Yes               1.0               1.0   

   Family History of Mental Illness  Depression  
2                                No           1  
8                               Yes           0  
26                              Yes           0  
30                              Yes           1  
32                               No           0

In [12]:
student_data[student_data['Profession'].isna()]


id       Name  Gender   Age           City  \
2            2     Yuvraj    Male  33.0  Visakhapatnam   
8            8  Aishwarya  Female  24.0      Bangalore   
26          26     Aditya    Male  31.0       Srinagar   
30          30     Prisha  Female  28.0       Varanasi   
32          32     Chhavi  Female  25.0         Jaipur   
...        ...        ...     ...   ...            ...   
140685  140685   Aaradhya  Female  27.0          Surat   
140686  140686      Rohan    Male  27.0       Ludhiana   
140689  140689      Ayaan    Male  31.0      Faridabad   
140690  140690      Rashi  Female  18.0       Ludhiana   
140699  140699        Sai    Male  27.0          Patna   

       Working Professional or Student Profession  Academic Pressure  \
2                              Student        NaN                5.0   
8                              Student        NaN                2.0   
26                             Student        NaN                3.0   
30                             Student        NaN                3.0   
32                             Student        NaN                4.0   
...                                ...        ...                ...   
140685                         Student        NaN                5.0   
140686                         Student        NaN                2.0   
140689                         Student        NaN                3.0   
140690                         Student        NaN                5.0   
140699                         Student        NaN                4.0   

        Work Pressure  CGPA  Study Satisfaction  Job Satisfaction  \
2                 NaN  8.97                 2.0               NaN   
8                 NaN  5.90                 5.0               NaN   
26                NaN  7.03                 5.0               NaN   
30                NaN  5.59                 2.0               NaN   
32                NaN  8.13                 3.0               NaN   
...               ...   ...                 ...               ...   
140685            NaN  5.75                 5.0               NaN   
140686            NaN  9.40                 3.0               NaN   
140689            NaN  6.61                 4.0               NaN   
140690            NaN  6.88                 2.0               NaN   
140699            NaN  9.24                 1.0               NaN   

           Sleep Duration Dietary Habits    Degree  \
2               5-6 hours        Healthy   B.Pharm   
8               5-6 hours       Moderate       BSc   
26      Less than 5 hours        Healthy        BA   
30              7-8 hours       Moderate       BCA   
32              5-6 hours       Moderate    M.Tech   
...                   ...            ...       ...   
140685          5-6 hours      Unhealthy  Class 12   
140686  Less than 5 hours        Healthy       MSc   
140689          5-6 hours      Unhealthy        MD   
140690  Less than 5 hours        Healthy  Class 12   
140699  Less than 5 hours        Healthy       BCA   

       Have you ever had suicidal thoughts ?  Work/Study Hours  \
2                                        Yes               3.0   
8                                         No               3.0   
26                                        No               9.0   
30                                       Yes               4.0   
32                                       Yes               1.0   
...                                      ...               ...   
140685                                   Yes               7.0   
140686                                    No               0.0   
140689                                    No              12.0   
140690                                   Yes              10.0   
140699                                   Yes               2.0   

        Financial Stress Family History of Mental Illness  Depression  
2                    1.0                               No           1  
8                    2.0           

In [13]:
student_data_irrelevant_columns = ['Working Professional or Student','Work Pressure','CGPA','Job Satisfaction','Profession']
student_data = student_data.drop(columns=student_data_irrelevant_columns,axis=1)

Look into this outlier data:

student_data[student_data['CGPA'].isna()].index

Index([21880, 52737, 58062, 70453, 75007, 105773, 110097, 129756, 134830], dtype='int64')


Profession is also filled out for a number of students

In [14]:
student_data.to_csv('../data_split_0/student_train_data.csv',index=False)